src/plugins/intel_npu/src/plugin/npuw/partitioning/patterns/dcoff.cpp

```diff
@@ -125,7 +126,11 @@ void apply_remap(Subgraph& fcall, const ClosureRemap& m) {
     // empty tensors by default.
     for (auto&& i : m.closure_remap) {
         new_lazy_closure.push_back(fcall._lazy_closure[i]);
-        new_closure.push_back(fcall._closure[i]);
+        if (fcall._closure[i]) {
+            new_closure.push_back(fcall._closure[i]);
+        } else {
+            new_closure.push_back(m.weights_to_unpack.count(i) ? fcall._lazy_closure[i].eval() : fcall._closure[i]);
+        }
```

Comment:
Trying to model this situation:

1. Imagine we have an `f16` weight (`Constant`) in our closure
2. When we initialize our function pipeline, this weight is turned into a stub `LazyClosure` - so is stored in `_lazy_closure`.
3. When we do DCOFF, we don't mark it as scale or zerop - so in `build_remap`, we mark store its index in `closure_remap` (as we did before) and, now, in `weights_to_unpack` (even if this weight doesn't in fact require unpack).
4. Now, in `apply_remap`:
   * `fcall._closure` is empty, so we enter `else`
   * in `else`, as `weights_to_unpack` lists this parameter, we'll take `lazy_closure` and evaluate it into a host-side tensor.

What it essentially means - if a model is full FP16, we'll pin all our FP16 tensors from a lazy closure into a normal closure and so won't put it through a bank.

That's okay-ish for now it it requires a proper fix - we _do need_ to mark only weights we know we'll unpack as `weights_to_unpack`. So as a follow-up, the logic that's written in `build_remap` needs to be extended.

Reply:
Well, this whole second fix was for host gather. We could implicitly set the index somewhere during partitioning, so here this first `if` would only check this index (could be set for future potential extensions besides host gather)

Reply:
Opened a ticket 142965